### PR TITLE
Majoron limits

### PR DESCRIPTION
**Pull request by ashleyrback**
Supersedes #56 

Changes include:

* Adds `echidna.__echidna_base` which points to base directory of echidna repo
* Adds to `constants`, including some SNO+ defaults
* Modifications to `decay` methods, mainly some re-naming and re-structuring. Also adds better handling of ROI efficiency.
* Adds physics tests in `decay.main`, which test compatibility against Andy's numbers (numbers from docDB)
* In `Spectra` class:
    * Adds `shrink_to_roi` which, given an axis, lower and upper limit for an ROI, calls the standard `shrink` method and also calculates the efficiency (`roi_integral / full_integral`) and stores the parameters in the dict `Spectra._rois[axis]`.
    * Adds getter/setter methods to save plotting style information in `Spectra._style`
    * Adds `cut` method which is similar to `shrink` but updates `Spectra._num_decays`, meaning that the `Spectra` object represents a different number of decays after calling this method.
* In `chi_squared`:
    * Adds `set_penalty_term` method
    * Chi squared calculations now calculate chi squared per bin, accessed via `get_chi_squared_per_bin`
* Adds `counts_down` and corresponding `get_last_bin_above` methods, plus a few other helpful methods.
* Adds lots more useful content to `SystAnalyser` class
* In `LimitSetting`:
    * Adds debugging tool which produces a catalogue of spectral plots via `matplotlib`'s`pdfPages`
    * Functionality to add a target spectrum (e.g. limit set by another experiment). This can be shown on (debugging) spectral plot(s) to compare limits.
* In `plot` module:
    * Adds `spectral_plot` method to produce spectral plot
    * Removes import of `pylab` in module
* Adds more useful plotting methods to `plot_chi_squared`, including "trun_on" plot and "push_pull" plot. These are all useful diagnostic tools that help visualise the data collected by the `SystAnalyser` class.
* Bug fix in `store.load/dump_ndarray`
* Adds (example) limit setting scripts for SNO+ Majoron modes and Majoron  modes in Xe136
* Adds `physics_tests` module

### CODE REVIEW

File: echidna/calc/decay.py [hunk 1]
```diff
@@ -1,69 +1,113 @@
...
+            loading = const._loading
+        if loading < 0. or loading > 1.:
             raise ValueError("Loading ranges from 0 to 1")
-        if self._abundance < 0. or self._abundance > 1.:
-            raise ValueError("Abundance ranges from 0 to 1")
-        mass_iso = self._atm_weight_iso*const._atomic_mass_unit  # kg/atom
-        mass_nat = self._atm_weight_nat*const._atomic_mass_unit  # kg/atom
-        mass_fraction = self._abundance*mass_iso/mass_nat
-        if not self._scint_density:
-            self._scint_density = const._scint_density
```
Comment (jwaterfield) [inline echidna/calc/decay.py]: Should this of been removed? scint density is in the constants as well

File: echidna/calc/decay.py [hunk 2]
```diff
@@ -1,69 +1,113 @@
...
-        if not self._fv_radius:
-            self._fv_radius = const._fv_radius
-        scint_mass = (4./3.)*numpy.pi*self._fv_radius**3*self._scint_density
-        active_mass = scint_mass*mass_fraction*self._loading
-        return active_mass/mass_iso
-
-    def get_activity(self, half_life, n_atoms):
+        if scint_density is None:  # use default from constants
```
Comment (jwaterfield) [inline echidna/calc/decay.py]: I think maybe n_atoms should be a class variable as its used by many functions and only really needs to be calculated once in the initialization

File: echidna/test/physics_tests.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    # Set defaults
+    if kwargs.get("tolerance"):
+        tolerance = kwargs.get("tolerance")
+    else:
+        tolerance = 0.001
+
+    # Evaluate function
+    observed = function(*args, **kwargs)
+
+    result = numpy.isclose(observed, expected, atol=tolerance*expected)
```
Comment (jwaterfield) [inline echidna/test/physics_tests.py]: Change this to allclose for the time being to work with old numpys for testing

**Comment by ashleyrback** [comment]
@jwaterfield docs should now be fine too. except for a couple of problems displaying maths in `decay` module. I can't quite figure out why as they were displaying fine before and haven't been changed.

**Comment by jwaterfield** [comment]
Ok. Happy and merging

**Pull request merged**